### PR TITLE
Update azblob with the latest azcore

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This tactially resolves the small number of breaking changes in azcore.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/zc_storage_error.go [hunk 1]
```diff
@@ -70,22 +73,32 @@ func handleError(err error) error {
 }
 
 // defunkifyStorageError is a function that takes the "funky" ResponseError and reduces it to a storageError.
-func defunkifyStorageError(responseError ResponseError) error {
-	if err, ok := responseError.Unwrap().(*StorageError); ok {
-		// errors.Unwrap(responseError.Unwrap())
+func defunkifyStorageError(responseError *azcore.ResponseError) error {
```
Comment (RickWinter) [inline sdk/storage/azblob/zc_storage_error.go:76]: Not critical for this change, but this functional should be renamed.

File: sdk/storage/azblob/zc_storage_error.go [hunk 2]
```diff
@@ -70,22 +73,32 @@ func handleError(err error) error {
...
+		goto Default
+	}
+	if len(body) > 0 {
+		if err := xml.Unmarshal(body, &storageError); err != nil {
+			goto Default
+		}
+	}
 
-		err.response = responseError.RawResponse()
+	// errors.Unwrap(responseError.Unwrap())
```
Comment (RickWinter) [inline sdk/storage/azblob/zc_storage_error.go:88]: If unwrapping is no longer necessary, then this line should be removed (not commented out).

Comment (jhendrixMSFT) [reply sdk/storage/azblob/zc_storage_error.go:88]: This was a preexisting artifact.  I can remove it, but I prefer to leave general code hygiene to the owning team.

File: sdk/storage/azblob/zt_test.go [hunk 3]
```diff
@@ -533,7 +533,8 @@ func blockIDIntToBase64(blockID int) string {
 func validateStorageError(_assert *assert.Assertions, err error, code StorageErrorCode) {
 	_assert.NotNil(err)
 	var storageError *StorageError
-	_assert.Equal(errors.As(err, &storageError), true)
+	// TOOD: this should really be require.Equal so that if it fails we don't try the next line which will panic
```
Comment (seankane-msft) [inline sdk/storage/azblob/zt_test.go:536]: Just a note, I change these to `require` (which you're correct we should be using over `assert`) in my PR to move to the new test proxy

**Review by seankane-msft — APPROVED** [review]
Agree with Rick's comments, otherwise looks good to me

File: sdk/storage/azblob/zc_storage_error.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/storage/azblob/zc_storage_error.go]: Fixed

File: sdk/storage/azblob/zc_storage_error.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/storage/azblob/zc_storage_error.go]: Fixed

**Comment by mohsha-msft** [comment]
`dev-azblob` has changes which we need to include in the upcoming release i.e. `v0.3.0`. Hence changes the destination branch.

**Review by mohsha-msft — APPROVED** [review]

**Pull request merged**